hogql_parser/parser.cpp

```diff
@@ -344,6 +350,9 @@ class HogQLParseTreeConverter : public HogQLParserBaseVisitor {
         int extend_code = X_PyList_Extend(flattened_queries, sub_select_queries);
         if (extend_code == -1) goto select_queries_loop_py_error;
         Py_DECREF(sub_select_queries);
+      } else if (is_ast_node_instance(query, "Placeholder")) {
```

Comment:
Same as in the Python version, this would be more readable if we did ` || is_ast_node_instance(query, "Placeholder")` in the first `if ()` branch. (I know readability is not the C++ code's strong suit, but that makes reining it in all the more important.)

BTW `is_ast_node_instance(query, "SelectQuery");` is called twice in this function, which is an oversight on my side. It's intentionally assigned to `is_select_query` so that we can check against the error value (`-1`), which is great for perfect error handling. However, two lines later repeat that call without the check…
These operations are very low risk, so I think in this case I suggest we remove lines 341 and 342. Because if were to run the same error checks for all branches in an `else if` fashion, we'd have to do quite some `if` nesting…